include/oneapi/dpl/internal/random_impl/bernoulli_distribution.h

```diff
@@ -0,0 +1,193 @@
+    }
+
+    void
+    param(const param_type& __param)
+    {
+        p_ = __param.p;
+    }
+
+    scalar_type
+    min() const
+    {
+        return false;
+    }
+
+    scalar_type
+    max() const
+    {
+        return true;
+    }
+
+    // Generate functions
+    template <class _Engine>
+    result_type
+    operator()(_Engine& __engine)
+    {
+        return operator()<_Engine>(__engine, param_type(p_));
+    }
+
+    template <class _Engine>
+    result_type
+    operator()(_Engine& __engine, const param_type& __params)
+    {
+        return generate<size_of_type_, _Engine>(__engine, __params);
+    }
+
+    template <class _Engine>
+    result_type
+    operator()(_Engine& __engine, unsigned int __random_nums)
+    {
+        return operator()<_Engine>(__engine, param_type(p_), __random_nums);
+    }
+
+    template <class _Engine>
+    result_type
+    operator()(_Engine& __engine, const param_type& __params, unsigned int __random_nums)
+    {
+        return result_portion_internal<size_of_type_, _Engine>(__engine, __params, __random_nums);
+    }
+
+  private:
+    // Size of type
+    static constexpr int size_of_type_ = internal::type_traits_t<result_type>::num_elems;
+
+    // Static asserts
+    static_assert(::std::is_same<scalar_type, bool>::value,
+                  "oneapi::dpl::bernoulli_distribution. Error: unsupported data type");
+
+    // Distribution parameters
+    double p_;
+
+    // Implementation for generate function
+    template <int _Ndistr, class _Engine>
+    typename ::std::enable_if<(_Ndistr != 0), result_type>::type
+    generate(_Engine& __engine, const param_type& __params)
+    {
+        return generate_vec<_Ndistr, _Engine>(__engine, __params);
+    }
+
+    // Specialization of the scalar generation
+    template <int _Ndistr, class _Engine>
+    typename ::std::enable_if<(_Ndistr == 0), result_type>::type
+    generate(_Engine& __engine, const param_type& __params)
+    {
+        oneapi::dpl::uniform_real_distribution<double> __distr;
+        return __distr(__engine) < __params.p;
+    }
+
+    // Specialization of the vector generation with size = [1; 3]
+    template <int __N, class _Engine>
+    typename ::std::enable_if<(__N <= 3), result_type>::type
+    generate_vec(_Engine& __engine, const param_type& __params)
+    {
+        return generate_n_elems<_Engine>(__engine, __params, __N);
+    }
+
+    // Specialization of the vector generation with size = [4; 8; 16]
+    template <int __N, class _Engine>
+    typename ::std::enable_if<(__N > 3), result_type>::type
+    generate_vec(_Engine& __engine, const param_type& __params)
+    {
+        oneapi::dpl::uniform_real_distribution<sycl::vec<double, __N>> __distr;
+        sycl::vec<double, __N> __u = __distr(__engine);
+        sycl::vec<long, __N> __res_long = __u < sycl::vec<double, __N>{__params.p};
+        result_type __res;
+        for (int i = 0; i < __N; i++)
+            __res[i] = static_cast<bool>(__res_long[i]);

```

Comment:
Can it be done by using convert() method of vector?

Reply:
No, it impossible for bool as I see.